src/clojure/nrepl/transport.clj

```diff
@@ -123,6 +130,40 @@
             (.close in)
             (.close out))))))))
 
+(defn stringify-everything
+  [m]
+  (walk/postwalk (fn [v] (if (keyword? v)
+                           (let [nspace (namespace v)
+                                 name      (name v)]
+                             (str (when nspace (str nspace "/")) name))
+                           v))
+                 m))
+
+(defn nrepl+edn
+  "Returns a Transport implementation that serializes messages
+   over the given Socket or InputStream/OutputStream using EDN."
+  ([^Socket s] (nrepl+edn s s s))
+  ([in out & [^Socket s]]
+   (let [in (java.io.PushbackReader. (io/reader in))
+         out (io/writer out)]
+     (fn-transport
+      #(rethrow-on-disconnection s (stringify-everything (edn/read in)))
```

Comment:
That was done mostly to appease the tests, right? In general it seems redundant to convert the keywords to strings.